awx/main/access.py

```diff
@@ -588,17 +588,41 @@ def can_delete(self, obj):
 
 
 class InstanceGroupAccess(BaseAccess):
+    """
+    I can see Instance Groups when I am:
+       - a superuser(system administrator)
+       - an organization admin
+       - at least read_role on the instance group
+    I can edit Instance Groups when I am:
+       - a superuser
+       - an organization admin
+       - admin role on the Instance group
+    I can add/delete Instance Groups:
+       - a superuser(system administrator)
+    """
+
     model = InstanceGroup
     prefetch_related = ('instances',)
 
     def filtered_queryset(self):
-        return InstanceGroup.objects.filter(organization__in=Organization.accessible_pk_qs(self.user, 'admin_role')).distinct()
+        if self.user.admin_of_organizations.exists():
+            return InstanceGroup.objects.filter(organization__in=Organization.accessible_pk_qs(self.user, 'admin_role')).distinct()
+        return self.model.accessible_objects(self.user, 'read_role')
+
+    @check_superuser
+    def can_use(self, obj):
+        return self.user in obj.use_role
```

Comment:
You have a criteria in `filtered_queryset`, which effectively sets "read" permission. This criteria allows org admins to see instance groups associated with their org (by a m2m relationship).

The point of this comment is to point out that these users (org admins) will not gain "use" permission, because the `use_role` here only has the system admin role as a parent role, not any org admin roles. So these users will be able to see those IGs, but will not be able to associate them to a JT, for example. I don't know if this is a problem or not.

Reply:
There is special logic around both inventory and jt can_attach that handles this case. I've left that logic in for backward compatibility.